wsclient/src/main/java/org/hpccsystems/ws/client/HPCCFileSprayClient.java

```diff
@@ -749,27 +759,36 @@ public boolean upLoadLandingZone(File uploadFile, DropZone dropZone)
             Utils.startMulti(output, uploadFile.getName(), boundary, "");
             InputStream input = new FileInputStream(uploadFile.getAbsolutePath());
 
-            ByteBuffer buffer = ByteBuffer.allocate(1000000);
+            ByteBuffer buffer = ByteBuffer.allocate(BUFFER_LENGTH * 16);
 
             RandomAccessFile aFile = new RandomAccessFile(uploadFile.getAbsolutePath(), "rw");
             FileChannel inChannel = aFile.getChannel();
             WritableByteChannel outchannel = Channels.newChannel(output);
-            while (inChannel.read(buffer) > 0)
+            try
             {
-                buffer.flip();
-                while (buffer.hasRemaining())
+                while (inChannel.read(buffer) > 0)
                 {
-                    outchannel.write(buffer);
+                    buffer.flip();
+                    while (buffer.hasRemaining())
+                    {
+                        outchannel.write(buffer);
+                    }
                 }
             }
-
-            Utils.closeMulti(output, boundary);
-            outchannel.close();
-            inChannel.close();
-            output.close();
-            input.close();
-            aFile.close();
-            // server response
+            catch (IOException e){
+                e.printStackTrace();
+                returnValue = false;
+            }
+            finally {
```

Comment:
It depends in which scope you declare the channels, if you need to, they can be declared outside the outer try. 
Bottom line is if line 766 or others throw an exception, the channels are not closed.